src/state/util.ts

```diff
@@ -1,20 +1,23 @@
 import { find } from "lodash";
-import { AnyAction, Reducer } from "redux";
+import type { Action, Reducer } from "redux";
 
 import { APP_ID, CELL_ID_KEY } from "../constants";
 
 import { DiscreteMeasuredFeatureDef, FileInfo, MeasuredFeatureDef } from "./metadata/types";
 import { BatchedAction, TypeToDescriptionMap } from "./types";
 
-export function makeConstant(associatedReducer: string, actionType: string) {
-    return `${APP_ID}/${associatedReducer.toUpperCase()}/${actionType.toUpperCase()}`;
+export function makeConstant<R extends string, A extends string>(
+    associatedReducer: R,
+    actionType: A
+): `${typeof APP_ID}/${Uppercase<R>}/${Uppercase<A>}` {
+    return `${APP_ID}/${associatedReducer.toUpperCase() as Uppercase<R>}/${actionType.toUpperCase() as Uppercase<A>}`;
```

Comment:
This change is not strictly required by the new types, but adds a fun bonus: action type constants now show up with their actual value in Intellisense. (e.g. the constant `CHANGE_GROUP_BY_CATEGORY` used to show up as having type `string` when you hovered over it; now it has type `"cell-feature-explorer/SELECTION/CHANGE-GROUP-BY-CATEGORY"`).